File: bench/bench.ts
import { cheerio, ConfigFactory, extract } from "@purescraps/purehtml";
import { readFileSync } from "fs";

const iterations = 10000;
const html = readFileSync("files/sample-1.html").toString();
const config = ConfigFactory.fromYAML(
  readFileSync("files/sample-1.yaml").toString()
);
const doc = cheerio.load(html);

console.log(`Running benchmark for ${iterations} iterations.`);

console.log(
  `extract result: ${extract(cheerio, doc, config, "http://example.com")}`
);

const start = Date.now();
for (let i = 0; i < iterations; i++) {
  extract(cheerio, doc, config, "https://example.com");
}
const duration = Date.now() - start;
const throughput = (iterations * 1000) / duration;

console.log(`Throughput: ${throughput.toFixed(2)} extracts/second`);

File: bench/bench.py
from bs4 import BeautifulSoup
from timeit import default_timer as timer
from purehtml import ConfigFactory, extract


def read_file(path: str) -> str:
    file = open(path, "r")
    content = file.read()
    file.close()

    return content


iterations = 5000
config = ConfigFactory.from_yaml(read_file("files/sample-1.yaml"))
html = read_file("files/sample-1.html")
doc = BeautifulSoup(html, "html5lib")

print(f"Running benchmark for {iterations} iterations.")

print(f"extract result: {extract(config, doc, "http://example.com")}")

start = timer()
for i in range(0, iterations):
    extract(config, doc, "http://example.com")
duration = timer() - start

throughput = iterations / duration

print(f"Throughput: {throughput:.2f} extracts/second")
